server/controllers/componentsController.test.ts

```diff
@@ -110,224 +44,95 @@ const defaultFooterViewModel = {
+    it('should return the HeaderViewModel for a non-prison user', async () => {
+      const output = await controller.getHeaderViewModel(hmppsUserMock)
+      expect(output).toEqual({ ...expectedHeaderViewModel, isPrisonUser: false })
     })
   })
-})
 
-describe('getFooterViewModel', () => {
-  it('should return the FooterViewModel with links from contentful if flag is true', async () => {
-    config.contentfulFooterLinksEnabled = true
-    const output = await controller.getFooterViewModel({
-      ...defaultTokenData,
-      authSource: 'nomis',
-      token: 'token',
-      roles: [],
-    })
-    expect(output).toEqual({
-      managedPages: [
-        { href: 'url1', text: 'text1' },
-        { href: 'url2', text: 'text2' },
-      ],
-      isPrisonUser: true,
-      component: 'footer',
-      services: defaultUserData.services,
+  describe('getFooterViewModel', () => {
+    it('should return the FooterViewModel with links from contentful if flag is true', async () => {
+      config.contentfulFooterLinksEnabled = true
+      const output = await controller.getFooterViewModel(prisonUserMock)
+      expect(output).toEqual({
+        managedPages: [
+          { href: 'url1', text: 'text1' },
+          { href: 'url2', text: 'text2' },
+        ],
+        isPrisonUser: true,
+        component: 'footer',
+      })
     })
-  })
 
-  it('should return the FooterViewModel with default links if flag is false', async () => {
-    config.contentfulFooterLinksEnabled = false
-    const output = await controller.getFooterViewModel({
-      ...defaultTokenData,
-      authSource: 'nomis',
-      token: 'token',
-      roles: [],
+    it('should return the FooterViewModel with default links if flag is false', async () => {
+      config.contentfulFooterLinksEnabled = false
+      const output = await controller.getFooterViewModel(prisonUserMock)
+      expect(output).toEqual(expectedFooterViewModel)
     })
-    expect(output).toEqual(defaultFooterViewModel)
-    expect(userServiceMock.getUserData).toBeCalledTimes(1)
-  })
 
-  describe('user data is passed in', () => {
-    it('should use the data passed in and not call userService', async () => {
+    it('should return the FooterViewModel for non-prison users', async () => {
       config.contentfulFooterLinksEnabled = false
-      const output = await controller.getFooterViewModel(
-        {
-          ...defaultTokenData,
-          authSource: 'nomis',
-          token: 'token',
-          roles: [],
-        },
-        defaultUserData,
-      )
-      expect(output).toEqual(defaultFooterViewModel)
-      expect(userServiceMock.getUserData).toBeCalledTimes(0)
+      const output = await controller.getFooterViewModel(hmppsUserMock)
+      expect(output).toEqual({ ...expectedFooterViewModel, isPrisonUser: false })
     })
   })
-})
 
-describe('getViewModels', () => {
-  it('should get user data if nothing in cache', async () => {
```

Comment:
the controller is no longer concerned with the cached data, this is handled by the userService completely now